src/dex-ui-components/presets/SwapTokensForm/SwapTokensForm.tsx

```diff
@@ -60,42 +60,56 @@ export function SwapTokensForm(props: SwapTokensFormProps) {
     },
   });
   const formValues: SwapTokensFormData = structuredClone(swapTokensForm.getValues());
-  swapTokensForm.watch("tokenToTrade.displayAmount");
-  swapTokensForm.watch("tokenToTrade.symbol");
-  swapTokensForm.watch("tokenToReceive.symbol");
-  const selectedRoute = {
-    selectedPairId: formValues.tokenToReceive.tokenMeta.pairAccountId,
-    selectedAToBRoute: `${formValues.tokenToTrade.tokenMeta.tokenId}=>${formValues.tokenToReceive.tokenMeta.tokenId}`,
-    selectedBToARoute: `${formValues.tokenToReceive.tokenMeta.tokenId}=>${formValues.tokenToTrade.tokenMeta.tokenId}`,
-  };
-  useSwapData(selectedRoute, REFRESH_INTERVAL);
+  /**
+   * Changes to these form inputs require the form to rerender.
+   */
+  swapTokensForm.watch([
+    "tokenToTrade.displayAmount",
+    "tokenToTrade.symbol",
+    "tokenToReceive.symbol",
+    "tokenToReceive.displayAmount",
+    "tokenToTrade.poolLiquidity",
+    "tokenToReceive.poolLiquidity",
+  ]);
+
+  const selectedPairContractId = formValues.tokenToReceive.tokenMeta.pairAccountId ?? "";
+  useSwapData(selectedPairContractId, REFRESH_INTERVAL);
   const formSettings = useFormSettings({ initialSlippage: formValues.slippage });
 
   const successMessage = `Swapped
-  ${formValues.tokenToTrade.amount.toFixed(6)} 
+  ${formValues.tokenToTrade.amount.toFixed(8)} 
   ${formValues.tokenToTrade.symbol}
-for ${formValues.tokenToReceive.amount.toFixed(6)} ${formValues.tokenToReceive.symbol}`;
+for ${formValues.tokenToReceive.amount.toFixed(8)} ${formValues.tokenToReceive.symbol}`;
 
   const notification = useNotification({ successMessage, transactionState: props.transactionState });
 
-  const isSubmitButtonDisabled =
-    isEmpty(formValues.tokenToTrade.displayAmount) ||
-    isNil(formValues.tokenToTrade.symbol) ||
-    isEmpty(formValues.tokenToReceive.displayAmount) ||
-    isNil(formValues.tokenToReceive.symbol);
-
   const spotPrice = getSpotPrice({
     spotPrices: props.spotPrices,
     tokenToTrade: formValues.tokenToTrade,
     tokenToReceive: formValues.tokenToReceive,
   });
 
-  const priceImpact = getPriceImpact({
-    spotPrice,
-    tokenToTrade: formValues.tokenToTrade,
-    tokenToReceive: formValues.tokenToReceive,
+  const priceImpact = calculatePriceImpact({
```

Comment:
I updated this function to only accept necessary parameters instead of the entire TokenState.